src/gateio_new_coins_announcements_bot/rotating_proxy.py

```diff
@@ -0,0 +1,108 @@
+import itertools
+import socket
+import struct
+import threading
+import time
+from typing import Callable
+
+import requests
+
+import gateio_new_coins_announcements_bot.globals as globals
+from gateio_new_coins_announcements_bot.logger import logger
+
+
+_proxy_list = {}
+_proxy = None
+_event = threading.Event()
+
+
+def init_proxy():
+    threading.Thread(target=lambda: _every(60 * 10, _fetch_proxies)).start()
+    # Required for populating the proxy list when starting bot
+    _fetch_proxies()
+
+
+def _fetch_proxies():
+    logger.info("Fetching proxies...")
+    _proxy_list
+    global _proxy
+    threads: list[threading.Thread] = []
+    try:
+        proxy_res = requests.get(
+            "https://www.proxyscan.io/api/proxy?last_check=180&limit=20&type=socks5&format=txt&ping=1000"
```

Comment:
we might also wanna consider a different last_check value, but thats something we can adjust after more testing as well.
(For reference: https://www.proxyscan.io/api)